### ISSUE TITLE
Small Data issue

### ISSUE THREAD

**Issue by wafa-bouzouita** [hukaixuan19970627/yolov5_obb#342]
In my application I am limited to small data (size<50). I trained yolov5_obb using both transfer learning and from scratch (freeze=0, 10 or 24) and I obtained on all cases (mAP < 0.4 on training data) .

Please do you have an idea how to deal with small data with yolov5_obb? What modifications could ameliorate results? 

Thank you

**Comment by hukaixuan19970627** [comment]
The information you're providing isn't comprehensive enough, can you upload these files?
![image](https://user-images.githubusercontent.com/72599120/170259940-abf9425d-b747-41ed-9360-29a0381aff26.png)

**Comment by wafa-bouzouita** [comment]
Thank you for your response, I want to say that I am limited to train yolov5_obb with small data (training data size <50). But model performance I obtained is bad. For both methods : 
* method 1 :  I trained my data from scratch (freeze = 0 ) and obtained mAP <0.1 
* method 2 : I trained my data with transfer learning (freeze=10 or 24) I obtained mAP<0.4

**Comment by wafa-bouzouita** [comment]
![labels_correlogram](https://user-images.githubusercontent.com/56389034/170265330-57ea5ac9-2592-4477-b867-67ab25765e7a.jpg)
![labels_xyls](https://user-images.githubusercontent.com/56389034/170265354-4cdc3caf-3023-4811-92fa-950d2ba400d3.jpg)
![results](https://user-images.githubusercontent.com/56389034/170265357-38704dd8-9316-490d-9b64-48d956c37e0c.png)
![train_batch0](https://user-images.githubusercontent.com/56389034/170265360-adb3c226-1b7d-416c-90e1-1c3ef7ee8b9e.jpg)
![train_batch1](https://user-images.githubusercontent.com/56389034/170265363-1eb2749a-692f-479f-9ba4-0628845de22c.jpg)
![train_batch2](https://user-images.githubusercontent.com/56389034/170265364-85c884d1-5346-43a4-a46d-09c322f56421.jpg)
![val_batch0_labels](https://user-images.githubusercontent.com/56389034/170265365-a6f45cce-4731-49ec-bae8-a75a07c6e2b9.jpg)
![val_batch0_pred](https://user-images.githubusercontent.com/56389034/170265368-0659c3d2-3da5-43ad-95fe-99ec65641c92.jpg)

**Comment by wafa-bouzouita** [comment]
hyp.yaml contains this lines : 

lr0: 0.01
lrf: 0.2
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 0.05
cls: 0.5
cls_pw: 1.0
theta: 0.5
theta_pw: 1.0
obj: 1.0
obj_pw: 1.0
iou_t: 0.2
anchor_t: 4.0
fl_gamma: 0.0
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 180.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.5
fliplr: 0.5
mosaic: 0.95
mixup: 0.1
copy_paste: 0.0
cls_theta: 180
csl_radius: 2.0

**Comment by wafa-bouzouita** [comment]
opt.yaml contains this lines:

weights: /home/dev/Documents/yolov5_obb/yolov5m_finetune/weights/best.pt
cfg: ''
data: /home/dev/Documents/yolov5_obb/data/schneider_rotated.yaml
hyp: data/hyps/obb/hyp.yaml
epochs: 3000
batch_size: 16
imgsz: 600
rect: false
resume: false
nosave: false
noval: false
noautoanchor: false
evolve: null
bucket: ''
cache: null
image_weights: false
device: '0'
multi_scale: false
single_cls: false
adam: false
sync_bn: false
workers: 8
project: runs/train
name: exp
exist_ok: false
quad: false
linear_lr: false
label_smoothing: 0.0
patience: 100
freeze:
- 24
save_period: -1
local_rank: -1
entity: null
upload_dataset: false
bbox_interval: -1
artifact_alias: latest
save_dir: runs/train/exp55

**Comment by hukaixuan19970627** [comment]
From your loss curve, the model haven't converged, which means the accuracy of the model has the potential to increase.
![image](https://user-images.githubusercontent.com/72599120/170267381-ed624de3-a1e0-4842-8a58-b37e2d0230b7.png)
![image](https://user-images.githubusercontent.com/72599120/170267460-3fd9c584-1c5a-4567-84fb-5935ffd66a43.png)
So set --patience to 3000 to prevent earlystopping from happening.

**Comment by hukaixuan19970627** [comment]
Can you upload the results.png?

**Comment by hukaixuan19970627** [comment]
https://github.com/hukaixuan19970627/yolov5_obb/issues/367
